android-json-form-wizard/src/main/java/com/vijay/jsonwizard/rules/RulesEngineFactory.java

```diff
@@ -50,16 +53,72 @@ public RulesEngineFactory(Context context, Map<String, String> globalValues) {
     public RulesEngineFactory() {
     }
 
+    private Rules getDynamicRulesFromJsonArray(JSONArray jsonArray) {
+        try {
+            Rules rules = new Rules();
+            String key = jsonArray.optString(0);
```

Comment:
Maybe I'm getting this wrong but assuming the key is always at position `0` is not a universal truth and highly depends on how `jsonArrayRules` was created.

Anyone using the public methods that depends on this function would not immediately understand this. A better way might be to add the key as a `key:value` pair with a deterministic `key` name that can be a const defined somewhere. Then you can search for the `key` name in the json array.